### PR TITLE
Harvester config

### PR DESCRIPTION
**Pull request by noirbizarre**
This pull request implenent the harvester declaration and validation workflow.

Mapping editor is disabled until ready.

### CODE REVIEW

File: js/components/harvest/source.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
         p="{{ jobs }}"
         empty="{{ _('No job yet') }}">
         <header>
-            {{ source.description | markdown }}
+            {{{ source.description | markdown }}}
```
Comment (davidbgk) [inline js/components/harvest/source.vue]: 3 brackets?

File: js/components/notifications/validate-harvester.vue [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    {{ _('Pending harvester validation for __name__', details) }}
+</a>
+</template>
+
+<script>
+export default {
+    mixins: [require('components/notifications/base')],
+    methods: {
+        click: function() {
+            this.$go('/harvester/' + this.details.id + '#validate');
```
Comment (davidbgk) [inline js/components/notifications/validate-harvester.vue]: No trailing slash?

File: js/views/harvester-wizard.vue [hunk 3]
```diff
@@ -41,20 +41,20 @@ export default {
...
-                    if (component.$.form.validate()) {
-                        Object.assign(this.source, component.serialize());
-                        this.source.save();
-                        this.source.$once('updated', () => {
-                            this.$.wizard.go_next();
-                        });
-                        return false;
-                    }
-                }
+            // }, {
```
Comment (davidbgk) [inline js/views/harvester-wizard.vue]: Remove?

File: js/views/harvester-wizard.vue [hunk 4]
```diff
@@ -17,7 +17,7 @@ export default {
             steps: [{
                 label: this._('Harvest as'),
                 subtitle: this._('Choose who is harvesting'),
-                component: 'publish-as',
+                component: 'publishas',
```
Comment (davidbgk) [inline js/views/harvester-wizard.vue]: Should we generalize that?

File: migrations/2015-09-25-migrate-harvest-sources-validation.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+var sources = db.harvest_source.find({validated: {$exists: true}}),
+    counter = 0;
+
+sources.forEach(function(source) {
+    source.validation = {
+        state: source.validated ? 'accepted' : 'pending',
+        comment: source.validation_comment,
+        on: source.create_at
```
Comment (davidbgk) [inline migrations/2015-09-25-migrate-harvest-sources-validation.js]: `created_at` is probably more consistent?

File: udata/core/user/models.py [hunk 6]
```diff
@@ -108,7 +108,8 @@ def organizations(self):
 
     @property
     def sysadmin(self):
-        return bool(getattr(g, 'sysadmin'))
+        return self.has_role('admin')
+        # return bool(getattr(g, 'sysadmin'))
```
Comment (davidbgk) [inline udata/core/user/models.py]: To remove?

File: udata/harvest/models.py [hunk 7]
```diff
@@ -61,6 +61,36 @@ class HarvestItem(db.EmbeddedDocument):
...
+                           default=VALIDATION_PENDING,
+                           required=True)
+    by = db.ReferenceField('User')
+    on = db.DateTimeField()
+    comment = db.StringField()
+
+
+class HarvestSourceQuerySet(db.BaseQuerySet):
+    def owned_by(self, *owners):
+        Qs = db.Q()
```
Comment (davidbgk) [inline udata/harvest/models.py]: Lowercase var in Python?

File: js/components/harvest/source.vue (hunk 1 above)
Comment (noirbizarre) [inline js/components/harvest/source.vue]: Yep, it means: "Do not escape". markdown returns raw HTML.

File: js/components/notifications/validate-harvester.vue (hunk 2 above)
Comment (noirbizarre) [inline js/components/notifications/validate-harvester.vue]: I will add it.
I also need to remove the hash, it's not needed anymore.

File: js/views/harvester-wizard.vue (hunk 3 above)
Comment (noirbizarre) [inline js/views/harvester-wizard.vue]: It's just temporary, I'm working on it.

File: js/views/harvester-wizard.vue (hunk 4 above)
Comment (noirbizarre) [inline js/views/harvester-wizard.vue]: I think we should, I'm tryng to gain on the produced JS size.
Each char gained in the template is welcome :)

Comment (davidbgk) [inline js/views/harvester-wizard.vue]: https://xkcd.com/1022/

Comment (noirbizarre) [inline js/views/harvester-wizard.vue]: :)

**Comment by davidbgk** [comment]
r+ remove commented code though

**Pull request merged**